src/api-service/__app__/timer_retention/__init__.py

```diff
@@ -0,0 +1,94 @@
+import datetime
+import logging
+
+import azure.functions as func
+from onefuzztypes.enums import JobState, TaskState
+
+from ..onefuzzlib.events import get_events
+from ..onefuzzlib.jobs import Job
+from ..onefuzzlib.notifications.main import Notification
+from ..onefuzzlib.repro import Repro
+from ..onefuzzlib.tasks.main import Task
+
+RETENTION_POLICY = datetime.timedelta(minutes=(5))
+SEARCH_EXTENT = datetime.timedelta(minutes=(15))
+
+
+def main(mytimer1: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa: F841
+
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+
+    time_retained_older = now - RETENTION_POLICY
+    time_retained_newer = now - SEARCH_EXTENT
+
+    time_filter = (
+        f"Timestamp lt datetime'{time_retained_older.isoformat()}' and "
+        f"Timestamp gt datetime'{time_retained_newer.isoformat()}'"
+    )
+    time_filter_newer = f"Timestamp gt datetime'{time_retained_older.isoformat()}"
+
+    # Collecting 'still relevant' task containers.
+    used_containers = []

```

Comment:
If you have 3 tasks that use container `foo`, then this will be `[Container("foo"), Container("foo"), Container("foo")]`

If you use a set, this will be: `{Container("foo")}`

Reply:
Good point. Addressing.